src/main/java/Sb2.java

```diff
@@ -131,5 +132,24 @@ public int getScriptCountForSprite(String spriteName) {
     public int[] getScriptLengthsForSprite(String spriteName) {
         return sprites.getScriptLengthsForSprite(spriteName);
     }
-}
+    /**
+     * Gets the number of global variables within the stage object.
+     * @return numVariables The number of global variables
+     */
+    public int getGlobalVariableCount() {
```

Comment:
For the purposes of this comment, I will use `gV` to refer to the variable called `globalVariables`.

@thehunte199, I think we can accomplish this task with one public method and no instance variables.  Currently, you have two methods for this task and `gV` is set up as in instance variable.  We don't need to retain `gV` as part of the state of the Sb2 instance.  

I would get rid of `public int getGlobalVariableCount()` and change `private void countGlobalVariables()` to a public method that returns an int.  Then you can make `gV` a local variable inside of `countGlobalVariables()`.  Does that make any sense?